pkg/capi/aws.go

```diff
@@ -0,0 +1,266 @@
+		awsMachineTemplate = newAWSMachineTemplate(mapiDefaultProviderSpec)
+		if err = cl.Create(ctx, awsMachineTemplate); err != nil {
+			Expect(err).ToNot(HaveOccurred(), "Failed to create awsmachinetemplate")
+		}
+		machineSetParams = framework.UpdateCAPIMachineSetName("aws-machineset-51071", machineSetParams)
+		machineSet, err = framework.CreateCAPIMachineSet(ctx, cl, machineSetParams)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset")
+		framework.WaitForCAPIMachinesRunning(ctx, cl, machineSet.Name)
+	})
+
+	//huliu-OCP-75395 - [CAPI] AWS Placement group support.
+	It("should be able to run a machine with cluster placement group", func() {
+		awsClient := framework.NewAwsClient(framework.GetCredentialsFromCluster(oc))
+		placementGroupName := clusterName + "pgcluster"
+		placementGroupID, err := awsClient.CreatePlacementGroup(placementGroupName, "cluster")
+		Expect(err).ToNot(HaveOccurred(), "Failed to create placementgroup")
+		Expect(placementGroupID).ToNot(Equal(""), "expected the placementGroupID to not be empty string")
+		defer func() {
+			framework.DeleteCAPIMachineSets(ctx, cl, machineSet)
+			framework.WaitForCAPIMachineSetsDeleted(ctx, cl, machineSet)
+			framework.DeleteObjects(ctx, cl, awsMachineTemplate)
+			deleted = true
+			_, err = awsClient.DeletePlacementGroup(placementGroupName)
+			Expect(err).ToNot(HaveOccurred(), "Failed to delete placementgroup")
+		}()
+
+		awsMachineTemplate = newAWSMachineTemplate(mapiDefaultProviderSpec)
+		awsMachineTemplate.Spec.Template.Spec.PlacementGroupName = placementGroupName
+		if err = cl.Create(ctx, awsMachineTemplate); err != nil {
+			Expect(err).ToNot(HaveOccurred(), "Failed to create awsmachinetemplate")
+		}
+		machineSetParams = framework.UpdateCAPIMachineSetName("aws-machineset-75395", machineSetParams)
+		machineSet, err = framework.CreateCAPIMachineSet(ctx, cl, machineSetParams)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset")
+		framework.WaitForCAPIMachinesRunning(ctx, cl, machineSet.Name)
+	})
+
+	//huliu-OCP-75396 - [CAPI] Creating machines using KMS keys from AWS.
+	It("should be able to run a machine using KMS keys", func() {
+		awsMachineTemplate = newAWSMachineTemplate(mapiDefaultProviderSpec)
+		region := mapiDefaultProviderSpec.Placement.Region
+		if region != "us-east-1" && region != "us-east-2" {
+			Skip("Region is " + region + ", skip this test scenario because we only created kms key in us-east-1/us-east-2 region")
+		}
+		var key string
+		switch region {
+		case "us-east-1":
+			key = "arn:aws:kms:us-east-1:301721915996:key/c471ec83-cfaf-41a2-9241-d9e99c4da344"
+		case "us-east-2":
+			key = "arn:aws:kms:us-east-2:301721915996:key/c228ef83-df2c-4151-84c4-d9f39f39a972"
+		}
+		awskmsClient := framework.NewAwsKmsClient(framework.GetCredentialsFromCluster(oc))
+		_, err = awskmsClient.DescribeKeyByID(key)
+		if err != nil {
+			Skip(fmt.Sprintf("Skip because cannot get the key %v", err))
+		}
```

Comment:
Skipping because of errors is likely to mean this test gets skipped when we don't want it to. Is there a better way to skip tests when we are in the dev account, vs QE account? Perhaps using labels?

Reply:
skipped for errors is a temporary solution, using labels should be better way, but we don't have that labels at present, and that's what we are discussing on slack https://redhat-internal.slack.com/archives/GE2HQ9QP4/p1732789329032099, we need identify which cases run in DEV CI, which cases run in QE CI, then add labels for them, then update all the DEV jobs to only run the cases chosen for DEV CI. Currently all the cases in this repo are run in DEV CI.

Reply:
Thank you @JoelSpeed I added a new label `qe-account-only` and updated Makefile to exclude use cases with such labels. And all other comments also updated, PLAT again, thanks!

Reply:
Does this new label align with the plan @sunzhaohua2 has proposed?

Reply:
Yes, the meaning is the same, just the name is different, Zhaohua named it `qe-only` and I named it `qe-account-only`, I can update the name when we make a decision.

Reply:
rebased the code after Zhaohua's label pr https://github.com/openshift/cluster-api-actuator-pkg/pull/344, I think the pr is ready for merge, @JoelSpeed PTAL, thanks!